recipes/haproxy_phpbb.rb

```diff
@@ -0,0 +1,61 @@
+#
+# Cookbook:: osl-keepalived
+# Recipe:: haproxy_phpbb
+#
+# Copyright:: 2018, Oregon State University
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+node.default['osl-keepalived']['master'] = {
+  'lb1.phpbb.org' => true,
+  'lb2.phpbb.org' => false
+}
+
+node.default['osl-keepalived']['priority'] = {
+  'lb1.phpbb.org' => 200,
+  'lb2.phpbb.org' => 100
+}
+
+include_recipe 'osl-keepalived::default'
+
+secrets = data_bag_item('osl_keepalived', 'haproxy_phpbb')
+
+keepalived_vrrp_instance 'haproxy-phpbb-ipv4' do
+  master node['osl-keepalived']['master'][node['fqdn']]
+  interface node['network']['default_interface']
+  virtual_router_id 3
+  priority node['osl-keepalived']['priority'][node['fqdn']]
+  authentication auth_type: 'PASS', auth_pass: secrets['auth_pass']
+  virtual_ipaddress %w(140.211.15.244/24)
+  notifies :reload, 'service[keepalived]'
+end
+
+keepalived_vrrp_instance 'haproxy-phpbb-ipv6' do
+  master node['osl-keepalived']['master'][node['fqdn']]
+  interface node['network']['default_interface']
+  virtual_router_id 4
+  priority node['osl-keepalived']['priority'][node['fqdn']]
+  # Authentication isn't actually used with IPv6 (see https://tools.ietf.org/html/rfc5798#section-9)
+  # however, keepalived cookbook requires this to be defined
+  authentication auth_type: 'PASS', auth_pass: secrets['auth_pass']
+  virtual_ipaddress %w() # TODO: IPv6 for lb.phpbb.com
+  notifies :reload, 'service[keepalived]'
+  action :nothing # TODO: IPv6 for lb.phpbb.com
+end
+
```

Comment:
phpBB has expressed interest in having IPv6 for their load balancer. I've allocated reverse pointers for their IPv6, but there are more steps before it will be ready. I'm leaving the logic in so there will be less changes to make when we deploy that (just flipping actions and adding tests, mostly).